### ISSUE TITLE
If the option does not begin with dash, linking is not performed

### ISSUE THREAD

**Issue by RobertoBagnara** [sphinx-doc/sphinx#2225]
Please note that this happens with Sphinx 1.3.3, i.e., well after the closure of issue #1658 .
To reproduce:

$ unzip bug.zip
$ cd bug
$ make html
$ google-chrome _build/html/index.html

[bug.zip](https://github.com/sphinx-doc/sphinx/files/82924/bug.zip)

**Comment by RobertoBagnara** [comment]
With `make xml' one can see a possible cause: compare

```
                <desc_signature allnames="nodash" first="False" ids="cmdoption-APROGRAM-arg-nodash"><desc_name>nodash</desc_name><desc_addname></desc_addname></desc_signature>
```

with

```
                <desc_signature allnames="-dash" first="False" ids="cmdoption-APROGRAM-dash"><desc_name>-dash</desc_name><desc_addname></desc_addname></desc_signature>
```

The difference is in the "arg-" bit.

**Comment by tk0miya** [comment]
MEMO: related with #1750

**Comment by tk0miya** [comment]
Fixed at 7894f0b
Thank you for reporting
